File: and-bible/app/src/test/java/net/bible/service/sword/SwordDocumentFacadeTest.java
package net.bible.service.sword;

import net.bible.android.TestBibleApplication;
import net.bible.android.activity.BuildConfig;
import net.bible.service.download.FakeSwordBookFactory;

import org.crosswire.jsword.book.Book;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.annotation.Config;

import robolectric.MyRobolectricTestRunner;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

@RunWith(MyRobolectricTestRunner.class)
@Config(constants = BuildConfig.class, application = TestBibleApplication.class)
public class SwordDocumentFacadeTest {

	private SwordDocumentFacade swordDocumentFacade;

	@Before
	public void setUp() throws Exception {
		swordDocumentFacade = new SwordDocumentFacade(null);
	}

	@Test
	public void testIsIndexDownloadAvailable() throws Exception {
		Book fakeBook = FakeSwordBookFactory.createFakeRepoBook("My Book", TestData.ESVS_CONF+"Version=1.0.1", "");
		assertThat(swordDocumentFacade.isIndexDownloadAvailable(fakeBook), equalTo(true));
	}

	interface TestData {
		String ESVS_CONF = "[ESVS]\nDescription=My Test Book\nCategory=BIBLE\nModDrv=zCom\nBlockType=CHAPTER\nLang=en\nEncoding=UTF-8\nLCSH=Bible--Commentaries.\nDataPath=./modules/comments/zcom/mytestbook/\n";
	}
}

File: app/src/test/java/net/bible/service/sword/SwordDocumentFacadeTest.kt
package net.bible.service.sword

import net.bible.service.download.FakeSwordBookFactory
import net.bible.service.download.RepoFactory
import org.crosswire.jsword.book.Book
import org.hamcrest.MatcherAssert
import org.hamcrest.core.IsEqual
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito

class SwordDocumentFacadeTest {
    private var swordDocumentFacade: SwordDocumentFacade? = null
    @Before
    fun setUp() {
		val repoFactory = Mockito.mock(RepoFactory::class.java)
        swordDocumentFacade = SwordDocumentFacade(repoFactory)
    }

    @Test
    @Throws(Exception::class)
    fun testIsIndexDownloadAvailable() {
        val fakeBook: Book = FakeSwordBookFactory.createFakeRepoBook("My Book", TestData.ESVS_CONF + "Version=1.0.1", "")
        MatcherAssert.assertThat(swordDocumentFacade!!.isIndexDownloadAvailable(fakeBook), IsEqual.equalTo(true))
    }

    internal interface TestData {
        companion object {
            const val ESVS_CONF = "[ESVS]\nDescription=My Test Book\nCategory=BIBLE\nModDrv=zCom\nBlockType=CHAPTER\nLang=en\nEncoding=UTF-8\nLCSH=Bible--Commentaries.\nDataPath=./modules/comments/zcom/mytestbook/\n"
        }
    }
}
